client/mysqlbinlog.cc

```diff
@@ -1005,7 +1008,8 @@ static bool print_row_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
   events to process has been reached and the program should terminate.
 */
 Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
```

Comment:
Actually `ev_end_pos` is (always) `pos + ev->data_written`. The signature change can be converted into  `assert(old_off + ev->data_written == my_b_tell(file))` before [process_event(...)](https://github.com/MariaDB/server/blob/review-10.5-Modev-35694/client/mysqlbinlog.cc#L2956).